Change setup.py to respect the SYSROOT environment variable.
This is needed to build the installed standard modules (e.g. readline & zlib).

--- a/setup.py
+++ b/setup.py
@@ -77,6 +77,7 @@ def get_platform():
     return sys.platform
 
 
+GENTOO_SYSROOT = os.environ.get("SYSROOT", "")
 CROSS_COMPILING = ("_PYTHON_HOST_PLATFORM" in os.environ)
 HOST_PLATFORM = get_platform()
 MS_WINDOWS = (HOST_PLATFORM == 'win32')
@@ -162,6 +163,15 @@ def sysroot_paths(make_vars, subdirs):
       headers or libraries.
     """
 
+    if GENTOO_SYSROOT:
+        dirs = []
+        for subdir in subdirs:
+            subdir.lstrip(os.sep)
+            path = os.path.join(GENTOO_SYSROOT, subdir)
+            if os.path.isdir(path):
+                dirs.append(path)
+        return dirs
+
     dirs = []
     for var_name in make_vars:
         var = sysconfig.get_config_var(var_name)
@@ -848,8 +858,8 @@ def configure_compiler(self):
         # directories (i.e. '.' and 'Include') must be first.  See issue
         # 10520.
         if not CROSS_COMPILING:
-            add_dir_to_list(self.compiler.library_dirs, '/usr/local/@@GENTOO_LIBDIR@@')
-            add_dir_to_list(self.compiler.include_dirs, '/usr/local/include')
+            add_dir_to_list(self.compiler.library_dirs, GENTOO_SYSROOT + '/usr/local/@@GENTOO_LIBDIR@@')
+            add_dir_to_list(self.compiler.include_dirs, GENTOO_SYSROOT + '/usr/local/include')
         # only change this for cross builds for 3.3, issues on Mageia
         if CROSS_COMPILING:
             self.add_cross_compiling_paths()
@@ -865,9 +875,9 @@ def init_inc_lib_dirs(self):
             # building a framework with different architectures than
             # the one that is currently installed (issue #7473)
             add_dir_to_list(self.compiler.library_dirs,
-                            sysconfig.get_config_var("LIBDIR"))
+                            GENTOO_SYSROOT + sysconfig.get_config_var("LIBDIR"))
             add_dir_to_list(self.compiler.include_dirs,
-                            sysconfig.get_config_var("INCLUDEDIR"))
+                            GENTOO_SYSROOT + sysconfig.get_config_var("INCLUDEDIR"))
 
         system_lib_dirs = ['/lib64', '/usr/lib64', '/lib', '/usr/lib']
         system_include_dirs = ['/usr/include']
@@ -1109,11 +1119,11 @@ def detect_readline_curses(self):
             elif curses_library:
                 readline_libs.append(curses_library)
             elif self.compiler.find_library_file(self.lib_dirs +
-                                                     ['/usr/@@GENTOO_LIBDIR@@/termcap'],
+                                                     [GENTOO_SYSROOT + '/usr/@@GENTOO_LIBDIR@@/termcap'],
                                                      'termcap'):
                 readline_libs.append('termcap')
             self.addext(Extension('readline', ['readline.c'],
-                               library_dirs=['/usr/@@GENTOO_LIBDIR@@/termcap'],
+                               library_dirs=[GENTOO_SYSROOT + '/usr/@@GENTOO_LIBDIR@@/termcap'],
                                libraries=readline_libs))
         else:
             self.addext(Extension('readline', ['readline.c']))
@@ -1126,7 +1136,7 @@ def detect_readline_curses(self):
         if curses_library == 'ncursesw':
             curses_defines.append(('HAVE_NCURSESW', '1'))
             if not CROSS_COMPILING:
-                curses_includes.append('/usr/include/ncursesw')
+                curses_includes.append(GENTOO_SYSROOT + '/usr/include/ncursesw')
             # Bug 1464056: If _curses.so links with ncursesw,
             # _curses_panel.so must link with panelw.
             panel_library = 'panelw'
